In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
import numpy as np  
#this is a python magic function to automatically show matplotlib plots
%matplotlib inline


In [2]:
evals_df = pd.read_csv('../data/bdi3_evals.csv', low_memory=False)
evals_df.head()

Child ID Gender Date of Birth Location - Sub Level 1  \
0    44879      F     9/27/2022              Southwest   
1    47795      F    12/23/2020      Greater Nashville   
2    54340      M     6/22/2021        First Tennessee   
3    54344      F      3/7/2022    Southeast Tennessee   
4    54515      M     11/4/2021          Memphis Delta   

                  Program Label  Adaptive Sum of Scaled Scores  \
0  BDI-3 Eligibility Evaluation                            7.0   
1  BDI-3 Eligibility Evaluation                           16.0   
2  BDI-3 Eligibility Evaluation                            8.0   
3  BDI-3 Eligibility Evaluation                           13.0   
4  BDI-3 Eligibility Evaluation                            5.0   

   Adaptive Developmental Quotient Adaptive Percentile Rank  \
0                             85.0                       16   
1                             88.0                       21   
2                             90.0                       25   
3                            115.0                       84   
4                             75.0                        5   

  Adaptive 95% Confidence Interval Adaptive RDI  ... Code 1  Code 2 Code 3  \
0                            77-97        39/90  ...    NaN     NaN    NaN   
1                            82-96        79/90  ...    NaN     NaN    NaN   
2                           81-101        77/90  ...    NaN     NaN    NaN   
3                          103-123        99/90  ...    NaN     NaN    NaN   
4                            68-88       Sep-90  ...    NaN     NaN    NaN   

   Code 4  Code 5 Code 6 Code 7 Code 8 Code 9  Code 10  
0     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
1     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
2     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
3     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
4     NaN     NaN    NaN    NaN    NaN    NaN      NaN  

[5 rows x 219 columns]

In [3]:
evals_df.tail()

Child ID Gender Date of Birth Location - Sub Level 1  \
23942  SP03287      F     9/25/2023          Memphis Delta   
23943  SP03288      F     5/18/2021          South Central   
23944  SP03289      F    10/31/2021          South Central   
23945  SP03290      M     3/25/2020          Memphis Delta   
23946  SP03291      M     9/18/2023      Greater Nashville   

                      Program Label  Adaptive Sum of Scaled Scores  \
23942  BDI-3 Eligibility Evaluation                           13.0   
23943  BDI-3 Eligibility Evaluation                           25.0   
23944  BDI-3 Eligibility Evaluation                           21.0   
23945       BDI-3 Annual Evaluation                           16.0   
23946  BDI-3 Eligibility Evaluation                           11.0   

       Adaptive Developmental Quotient Adaptive Percentile Rank  \
23942                            115.0                       84   
23943                            115.0                       84   
23944                            103.0                       58   
23945                             88.0                       21   
23946                            105.0                       63   

      Adaptive 95% Confidence Interval Adaptive RDI  ... Code 1  Code 2  \
23942                          103-123        99/90  ...    NaN     NaN   
23943                          107-121        96/90  ...    NaN     NaN   
23944                           96-110        92/90  ...    NaN     NaN   
23945                            82-96        66/90  ...    NaN     NaN   
23946                           94-114        94/90  ...    NaN     NaN   

      Code 3  Code 4  Code 5 Code 6 Code 7 Code 8 Code 9  Code 10  
23942    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
23943    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
23944    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
23945    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  
23946    NaN     NaN     NaN    NaN    NaN    NaN    NaN      NaN  

[5 rows x 219 columns]

In [4]:
evals_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23947 entries, 0 to 23946
Columns: 219 entries, Child ID to Code 10
dtypes: float64(79), object(140)
memory usage: 40.0+ MB


In [5]:
evaluations=evals_df.drop(columns=['Code 1','Code 2','Code 3','Code 4','Code 5','Code 6','Code 7','Code 8', 'Code 9','Code 10'])

In [6]:
evaluations.shape

(23947, 209)

In [7]:
evaluations.isnull().sum().sort_values(ascending=False)

Adaptive-Personal Responsibility RDI     10834
Social Emotional-Peer Interaction RDI    10828
Motor-Perceptual Motor RDI               10827
Motor-Perceptual Motor T-Score           10821
Motor-Perceptual Motor RS                10821
                                         ...  
Adaptive-Self Care NCE                       7
Gender                                       0
Program Label                                0
Date of Birth                                0
Child ID                                     0
Length: 209, dtype: int64

In [8]:
evaluations['Gender'].unique()

array(['F', 'M', 'U'], dtype=object)

In [9]:
evaluations[evaluations['Gender']=='U']

Child ID Gender Date of Birth Location - Sub Level 1  \
258     486118      U     3/30/2020         East Tennessee   
259     486118      U     3/30/2020         East Tennessee   
344     487659      U     7/29/2020         East Tennessee   
345     487659      U     7/29/2020         East Tennessee   
383     488355      U     8/24/2020         East Tennessee   
...        ...    ...           ...                    ...   
23712  SP03057      U      4/9/2021         East Tennessee   
23725  SP03070      U    12/18/2021         East Tennessee   
23793  SP03138      U     5/23/2023        First Tennessee   
23806  SP03151      U      7/9/2021         East Tennessee   
23809  SP03154      U     5/31/2022        First Tennessee   

                      Program Label  Adaptive Sum of Scaled Scores  \
258         BDI-3 Annual Evaluation                           21.0   
259         BDI-3 Annual Evaluation                           21.0   
344         BDI-3 Annual Evaluation                           23.0   
345         BDI-3 Annual Evaluation                           23.0   
383         BDI-3 Annual Evaluation                           18.0   
...                             ...                            ...   
23712  BDI-3 Eligibility Evaluation                           20.0   
23725  BDI-3 Eligibility Evaluation                            6.0   
23793  BDI-3 Eligibility Evaluation                            7.0   
23806  BDI-3 Eligibility Evaluation                            8.0   
23809  BDI-3 Eligibility Evaluation                            9.0   

       Adaptive Developmental Quotient Adaptive Percentile Rank  \
258                              103.0                       58   
259                              103.0                       58   
344                              109.0                       73   
345                              109.0                       73   
383                               94.0                       34   
...                                ...                      ...   
23712                            100.0                       50   
23725                             80.0                        9   
23793                             85.0                       16   
23806                             64.0                        1   
23809                             95.0                       37   

      Adaptive 95% Confidence Interval Adaptive RDI  ...  \
258                             96-110        90/90  ...   
259                             96-110        90/90  ...   
344                            101-115        94/90  ...   
345                            101-115        94/90  ...   
383                             87-101        82/90  ...   
...                                ...          ...  ...   
23712                           93-107        88/90  ...   
23725                            73-93        34/90  ...   
23793                            77-97        37/90  ...   
23806                            59-73        25/90  ...   
23809                           86-106        79/90  ...   

      Cognitive-Perception and Concepts RS  \
258                                   21.0   
259                                   21.0   
344                                   21.0   
345                                   21.0   
383                                   23.0   
...                                    ...   
23712                                 16.0   
23725                                 10.0   
23793                                  3.0   
23806                                 12.0   
23809                                  9.0   

       Cognitive-Perception and Concepts SS  \
258                                    10.0   
259                                    10.0   
344                                    11.0   
345                                    11.0   
383                                    12.0   
...                                     ...   
23712       

414 out of  23947 entries where gender = 'U' (~1.72%)

In [10]:
evaluations['Child ID'].nunique()

19003

There are 19003 unique child id.s for this data set(19003 children)

In [17]:
evaluations['Date of Birth'] = pd.to_datetime(evaluations['Date of Birth'], 
                                            format = '%m/%d/%Y')

In [18]:
evaluations.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23947 entries, 0 to 23946
Data columns (total 209 columns):
 #    Column                                                       Non-Null Count  Dtype         
---   ------                                                       --------------  -----         
 0    Child ID                                                     23947 non-null  object        
 1    Gender                                                       23947 non-null  object        
 2    Date of Birth                                                23947 non-null  datetime64[ns]
 3    Location - Sub Level 1                                       23815 non-null  object        
 4    Program Label                                                23947 non-null  object        
 5    Adaptive Sum of Scaled Scores                                23939 non-null  float64       
 6    Adaptive Developmental Quotient                              23939 non-null  float64       
 7    Ad

In [19]:
evaluations['Adaptive-Self Care Date of Testing'] = pd.to_datetime(evaluations['Adaptive-Self Care Date of Testing'], 
                                            format = '%m/%d/%Y')

In [20]:
evaluations['Adaptive-Personal Responsibility Date of Testing'] = pd.to_datetime(evaluations['Adaptive-Personal Responsibility Date of Testing'], 
                                            format = '%m/%d/%Y')

In [22]:
evaluations['Social Emotional-Adult Interaction Date of Testing'] = pd.to_datetime(evaluations['Social Emotional-Adult Interaction Date of Testing'], 
                                            format = '%m/%d/%Y')

In [23]:
evaluations['Social Emotional-Peer Interaction Date of Testing'] = pd.to_datetime(evaluations['Social Emotional-Peer Interaction Date of Testing'], 
                                            format = '%m/%d/%Y')

In [24]:
evaluations['Social Emotional-Self Concept / Social Role Date of Testing'] = pd.to_datetime(evaluations['Social Emotional-Self Concept / Social Role Date of Testing'], 
                                            format = '%m/%d/%Y')

In [25]:
evaluations['Communication-Receptive Communication Date of Testing'] = pd.to_datetime(evaluations['Communication-Receptive Communication Date of Testing'], 
                                            format = '%m/%d/%Y')

In [26]:
evaluations['Communication-Expressive Communication Date of Testing'] = pd.to_datetime(evaluations['Communication-Expressive Communication Date of Testing'], 
                                            format = '%m/%d/%Y')

Need to subtract date of testing minus DOB 